internal/beater/java_attacher/java_attacher.go

```diff
@@ -167,6 +177,71 @@ func (j *JavaAttacher) Run(ctx context.Context) error {
 	}
 }
 
+func (j *JavaAttacher) deleteTempDir(tempDir string) {
+	func(name string) {
+		err := os.RemoveAll(name)
+		if err != nil {
+			j.logger.Errorf("failed to delete Java attacher temp dir at %v: %v", tempDir, err)
+		}
+	}(tempDir)
+}
+
+// Creates a temp dir for the java attacher jar with the attacher jar in it, both with 0755 access permissions.
+// The assumption is that the default temp dir will be readable for all users.
+// However, there may be systems in which the temp directory is user-specific,
+// in which case we'll need to modify this function to run a temp dir search per user,
+// maybe by externally executing `echo XXX` and reading the output, relying on environment
+// variables as implemented in os.TempDir, or looking in known paths.
+// In such case, we would switch to using a cache of temp-agent-dir per user.
+func (j *JavaAttacher) createAttacherTempDir() (string, error) {
+	tempDir, err := os.MkdirTemp("", "elasticapmagent")
+	if err != nil {
+		return "", err
+	}
+	// todo: check if we need to use different value for specific OSs
```

Comment:
This comment was written because of the referred documentation. For example, once we support Windows, we may want the per-user attacher jar to have the `0200` bit off, which is not the case today.

Reply:
Irrelevant in light of what we now know about running on Windows (it's possible to attach to any JVM as the administrator)